src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,265 @@
+        /// If the broadcast is used after this is called, it will need to be re-sent to each
+        /// executor.
+        /// </summary>
+        public void Unpersist()
+        {
+            _jvmObject.Invoke("unpersist");
+        }
+
+        /// <summary>
+        /// Delete cached copies of this broadcast on the executors. If the broadcast is used after
+        /// this is called, it will need to be re-sent to each executor.
+        /// </summary>
+        /// <param name="blocking">Whether to block until unpersisting has completed</param>
+        public void Unpersist(bool blocking)
+        {
+            _jvmObject.Invoke("unpersist", blocking);
+        }
+
+        /// <summary>
+        /// Destroy all data and metadata related to this broadcast variable. Use this with
+        /// caution; once a broadcast variable has been destroyed, it cannot be used again.
+        /// This method blocks until destroy has completed.
+        /// </summary>
+        public void Destroy()
+        {
+            _jvmObject.Invoke("destroy");
+            File.Delete(_path);
+            JvmBroadcastRegistry.Remove(_jvmObject);
+        }
+
+        /// <summary>
+        /// Function that creates a file to store the broadcast value in the given path.
+        /// </summary>
+        /// <param name="tempDir">Path where file is to be created</param>
+        /// <param name="value">Broadcast value to be written to the file</param>
+        private void WriteBroadcastValueToFile(string tempDir, object value)
+        {
+            Directory.CreateDirectory(tempDir);
+            using FileStream f = File.Create(_path);
+            Dump(value, f);
+        }
+
+        /// <summary>
+        /// Function that serializes and stores the object passed to the given Stream.
+        /// </summary>
+        /// <param name="value">Serializable object</param>
+        /// <param name="stream">Stream to which the object is serialized</param>
+        private void Dump(object value, Stream stream)
+        {
+            var formatter = new BinaryFormatter();
+            formatter.Serialize(stream, value);
+        }
+
+        /// <summary>
+        /// Calls the necessary functions to create org.apache.spark.broadcast.Broadcast object
+        /// for Spark versions 2.3.0 and 2.3.1 and returns the JVMObjectReference object.
+        /// </summary>
+        /// <param name="tempDir">Path where file is to be created</param>
+        /// <param name="sparkContext">JVMObjectReference object of SparkContext</param>
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_1_Below(
+            string tempDir,
+            JvmObjectReference sparkContext,
+            object value)
+        {
+            WriteBroadcastValueToFile(tempDir, value);
+            var javaSparkContext = (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
```

Comment:
let's just use `_jvmObject` instead of `sparkContext.Jvm`.

Reply:
`_jvmObject` is not set yet, it is set as a result of calling these functions. Or am I missing something?